### PR TITLE
[DeepSeek-V4][DSpark] Keep main_proj in BF16

### PR DESCRIPTION
**Pull request by caozuoba**
## Summary

Fix low DSpark acceptance for DeepSeek-V4 W8A8 checkpoints.

`mtp.0.main_proj.weight` is stored in BF16 without quantization scales. Applying the target model's W8A8 quantization config to `main_proj` produces incorrect draft logits.

This change keeps `main_proj` unquantized with `quant_config=None`, restoring DSpark acceptance from single-digit rates to normal levels.

### CODE REVIEW

**Comment by ILikeIneine** [comment]
Hi, could you run pre-commit on this PR? It's failed.

```
# install pre-commit
pip install pre-commit 
pre-commit install

# then trigger the pre-commit by
git commit --amend --no-edit 
```

**Review by ILikeIneine — APPROVED** [review]
LGTM
